dcos/dcos-ui
# DCOS-55894 Support quota limits on group create

Support setting quota limits when creating a top-level group.
I also refactored the `ServiceRootGroupModal` to a directory to allow
modularizing it. The file was getting very large. And since we will want
to expand this modal even more in the future this seemed like a worth
while refactor now.

~*Note* this PR is based on #4000 and will need to be rebased once that PR merges.~

## Testing

In `Config.dev` set `useUIConfigFixtures: true` and 
```
  uiConfigurationFixture: {
    uiConfiguration: {
      features: {     
        quota: true
...
```

1. Go to `/services/overview`
2. Click `Create Group`
3. Verify that it works with and without quota.
4. Verify that the added tests make sense.

## Trade-offs



## Dependencies


~#4000~

## Screenshots

![image](https://user-images.githubusercontent.com/1972968/61981631-cded3500-afbf-11e9-899d-6b1b2b843033.png)
![image](https://user-images.githubusercontent.com/1972968/61981653-dd6c7e00-afbf-11e9-953e-46681a9aece7.png)
![image](https://user-images.githubusercontent.com/1972968/61981666-e65d4f80-afbf-11e9-9f9d-c35e690e17e9.png)
